### DISCUSSION TITLE
React-native Firebase Remote Config never use cache

### DISCUSSION THREAD

**Discussion by n10ty** [invertase/react-native-firebase#4781] [Q&A] [answered]
Remote config always fetching from server My setup remote-config:
```
async conf() {
    await remoteConfig().setDefaults({
      test_variable: 'default',
    });
    const fetchedRemotely = await remoteConfig().fetchAndActivate();

    if (fetchedRemotely) {
      console.log('Configs were retrieved from the backend and activated.');
    } else {
      console.log(
        'No configs were fetched from the backend, and the local configs were already activated',
      );
    }
  }
```

Fetching value:

```
const testVar = remoteConfig().getValue('test_variable');

    if (testVar.getSource() === 'remote') {
      console.log('Parameter value was from the Firebase servers.');
    } else if (testVar.getSource() === 'default') {
      console.log('Parameter value was from a default value.');
    } else {
      console.log('Parameter value was from a locally cached value.');
    }
```

I always getting (both: iOS and Android, also tried real device)

>Parameter value was from the Firebase servers.

so the source is from "remote", not from "cache" or "static" although it should cache for 12 hours by default. I can't do requests so often. What I did wrong? Any help?

**Comment by mikehardy** [comment, marked as answer]
Have you confirmed from the firebase console that you are indeed fetching more than expected? i.e., do you really have a problem?

I think that is just a misunderstanding of the source value.
It came from remote so I believe it will always say it is from remote even if it is not refetched
If the fetch never happened, I believe it would be from default or local
You don't indicate what the fetch methods return to you, if you carefully examine the return values of the various fetch / activate methods (and perhaps use the ones that are single actions, not multiple actions) you may find it is behaving as expected

**Reply by n10ty** [reply]
Thanks, Mike. Yes, I see fetches in FB console. 
The problem appears after I did several fetches and got blocked by firebase API because of too often requests (it was not really often: 5-10 per several minutes during development). So I'm afraid after deploying config to my app it will start doing enormous number of request and:
>Be warned Firebase may start to reject your requests if values are requested too frequently.

this will happen.
That's why I'm asking, I don't want any issues after start using it for real users.

**Reply by mikehardy** [reply]
Open a real issue with an App.js that is completely self-contained and reproduces the problem based on currrent versions so we can see what's going on. Perhaps you do not have the refresh time / minimum fetch interval configured correctly or are overriding it, as that would cause the same symptoms, and nothing else would cause those symptoms I think.

**Reply by n10ty** [reply]
@mikehardy Ok, I think you were right. I had a misunderstanding of the source value. I did a single App.js and it works as expected. Thanks.

**Reply by mikehardy** [reply]
Glad that worked for you! It is subtle stuff - each module is the size of normal non-firebase react-native module all with it's own stuff to learn. Good luck with your project!
